src/platform/plugins/shared/esql/server/services/esql_service.ts

```diff
@@ -67,16 +73,61 @@ export class EsqlService {
 
     const indices: IndexAutocompleteItem[] = [];
     const indexNames: string[] = [];
+    const indexNamesSet = new Set<string>(); // Track unique index names
 
     for (const [name, { settings }] of Object.entries(queryByIndexModeResponse)) {
       if (settings['index.mode'] === mode && !settings['index.hidden']) {
-        indexNames.push(name);
-        indices.push({ name, mode, aliases: [] });
+        if (!indexNamesSet.has(name)) {
+          indexNamesSet.add(name);
+          indexNames.push(name);
+          indices.push({ name, mode, aliases: [] });
+        }
       }
     }
 
     const aliases = await this.getIndexAliases(indexNames);
 
+    if (remoteClusters) {
+      const remoteClustersArray = remoteClusters.split(',');
+      // attach a wildcard * for each remoteCluster
+      const clustersArray = remoteClustersArray.map((cluster) => `${cluster.trim()}:*`);
+      const ccsSources = (await client.indices.resolveIndex({
+        name: clustersArray,
+        expand_wildcards: 'open',
+      })) as ResolveIndexResponse;
+
+      const cssIndicesForMode: string[] = [];
+      // ES will provide a filter for mode on the resolveIndex api so we can use this instead of
+      // filtering at the kibana side
+      ccsSources.indices?.forEach((index) => {
+        // filter by mode
+        if (index.mode === mode) {
+          // indices.push({ name: index.name, mode, aliases: [] });
+          cssIndicesForMode.push(index.name);
+        }
```

Comment:
Why not add here: 
```
const [clusterName] = index.name.split(":");
index.aliases?.forEach((alias) => {
  cssIndicesForMode.push(`$clusterName:$alias`);
}
```
or something similar? Then you don't need to parse `aliases` separately?